### PR TITLE
Support private extractor registration

### PR DESCRIPTION
**Pull request by max-zilla**
Adds support for a unique_key and owner parameter to register extractors for personal use. Links to come for related pyclowder and extractor code.

### CODE REVIEW

File: app/services/mongodb/MongoDBExtractorService.scala [hunk 1]
```diff
@@ -188,7 +188,7 @@ class MongoDBExtractorService @Inject() (
           case None => doc.permissions.isEmpty // If no user filter in registered extractor, everyone can see
         }
         val upper_categories = categories.map(cat => cat.toUpperCase)
-        val category_match = categories.length == 0 || doc.categories.intersect(upper_categories).length > 0
+        val category_match = doc.categories.length == 0 || doc.categories.intersect(upper_categories).length > 0
```
Comment (robkooper) [inline app/services/mongodb/MongoDBExtractorService.scala:191]: was this a bug?

**Review by robkooper — COMMENTED** [review]
LGTM, need to do testing

**Review by ddey2 — APPROVED** [review]
I tested it.Looks good to me

**Review by lmarini — CHANGES_REQUESTED** [review]
Max is adding checks to api.Extractions.submitFileToExtractor and api.Extractions.submitDatasetToExtractor

**Comment by robkooper** [comment]
not sure why the checks are not running, but on my local machine I get compile errors:

```
[error] /Users/kooper/git/clowder/clowder/app/controllers/Extractors.scala:44: not enough arguments for method getExtractorInfo: (extractorName: String, extractorKey: Option[String], user: Option[models.User])Option[models.ExtractorInfo].
[error] Unspecified value parameter user.
[error]     extractorService.getExtractorInfo(extractorName, extractor_key) match {
[error]                                      ^
```

```
[error] /Users/kooper/git/clowder/clowder/app/api/Extractions.scala:383: not enough arguments for method getExtractorInfo: (extractorName: String, extractorKey: Option[String], user: Option[models.User])Option[models.ExtractorInfo].
[error] Unspecified value parameter user.
[error]     extractors.getExtractorInfo(extractorName, extractor_key) match {
[error]                                ^
[error] /Users/kooper/git/clowder/clowder/app/api/Extractions.scala:525: not enough arguments for method getExtractorInfo: (extractorName: String, extractorKey: Option[String], user: Option[models.User])Option[models.ExtractorInfo].
[error] Unspecified value parameters extractorName, extractorKey, user.
[error]               extractors.getExtractorInfo()
[error]                                          ^
```

```
[error] /Users/kooper/git/clowder/clowder/app/services/ExtractorRoutingService.scala:79: not enough arguments for method getExtractorInfo: (extractorName: String, extractorKey: Option[String], user: Option[models.User])Option[models.ExtractorInfo].
[error] Unspecified value parameter user.
[error]       extractorsService.getExtractorInfo(exId, None)).filter(exInfo => {
[error]                                         ^
```

**Review by ddey2 — APPROVED** [review]

**Pull request merged**